src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -39,23 +65,21 @@ internal abstract partial class AbstractObjectBrowserLibraryManager : AbstractLi
         private object _classMemberGate = new object();
 
         private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _streamingPresenters;
+        private readonly Lazy<IComponentModel> _componentModel;
 
         protected AbstractObjectBrowserLibraryManager(
-            string languageName, 
-            Guid libraryGuid, 
-            __SymbolToolLanguage preferredLanguage, 
+            string languageName,
+            Guid libraryGuid,
+            __SymbolToolLanguage preferredLanguage,
             IServiceProvider serviceProvider)
             : base(libraryGuid, serviceProvider)
         {
             _languageName = languageName;
             _preferredLanguage = preferredLanguage;
+            _serviceProvider = serviceProvider;
 
-            var componentModel = (IComponentModel)serviceProvider.GetService(typeof(SComponentModel));
-            this.Workspace = componentModel.GetService<VisualStudioWorkspace>();
-            this.LibraryService = this.Workspace.Services.GetLanguageServices(languageName).GetService<ILibraryService>();
```

Comment:
:memo: As far as I can tell, all the value from this pull request would come from making `LibraryService` lazy.